lib/concentrate/parser/gtfs_realtime_enhanced.ex

```diff
@@ -152,21 +165,30 @@ defmodule Concentrate.Parser.GTFSRealtimeEnhanced do
     Map.get(vp, "occupancy_status")
   end
 
-  defp decode_state_of_charge(%{
-         "state_of_charge_percentage" => pct,
-         "state_of_charge_timestamp" => unix
-       })
+  defp decode_state_of_charge(
```

Comment:
suggestion: based on our conversation, I think this needs to include the case where the vehicle_id is in `@soc_vehicle_allow_list` but there's not SOC information, and return `%{time: nil, value: nil}`. Since the list is hardcoded, you could do something like
```elixir
defp decode_state_of_charge(vp, vehicle_id) when vehicle_id in @soc_vehicle_allow_list do
  case vp do
    %{
         "state_of_charge_percentage" => pct,
         "state_of_charge_timestamp" => unix
       } when is_integer(pct) and is_integer(unix) ->
        %{value: pct, time: unix}
    _ -> %{value: nil, time: nil}
  end
end

defp decode_state_of_charge(_vp, _vehicle_id) do
  nil
end
```